src/validation.cpp

```diff
@@ -835,33 +815,26 @@ bool MemPoolAccept::PreChecks(ATMPArgs& args, Workspace& ws)
                             newFeeRate.ToString(),
                             oldFeeRate.ToString()));
             }
+        }
+
+        // Calculate all conflicting entries and enforce Rule #5.
+        if (!GetEntriesForConflicts(tx, m_pool, setIterConflicting, allConflicting, err_string)) {
```

Comment:
f293c68be0469894c988711559f5528020c0ff71: I think for new code we should avoid leaking mutable return args into a greater scope than needed. Especially when the goal is to re-use (https://github.com/bitcoin/bitcoin/pull/22675/files) the symbol (err_string), when it shouldn't be re-used. With C++17 this is trivial to fix.

One example:

```cpp
         if (const auto err_string{GetEntriesForConflicts(tx, m_pool, setIterConflicting, allConflicting)}) {
            return ..., *err_string;
```

and make the util function return an error string or nullopt if no error is hit.

Re-using symbols has bitten us in ATMP in the past, so I don't think there is need to repeat that. See commit 116419e58dddef8fe3ff9806a1d8ceebe64ae3e6

Reply:
Good point, I've added a followup commit for `GetEntriesForConflicts` and edited the rest of the functions to return a `std::optional<std::string>` in #22675